csharp/sources/Valkey.Glide/Abstract/ConfigurationOptions.cs

```diff
@@ -0,0 +1,517 @@
+        }
+
+        internal const string
+            AbortOnConnectFail = "abortConnect",
+            AllowAdmin = "allowAdmin",
+            ConnectTimeout = "connectTimeout",
+            DefaultDatabase = "defaultDatabase",
+            KeepAlive = "keepAlive",
+            ClientName = "name",
+            User = "user",
+            Password = "password",
+            Proxy = "proxy",
+            ResponseTimeout = "responseTimeout",
+            Ssl = "ssl",
+            Version = "version",
+            SetClientLibrary = "setlib",
+            Protocol = "protocol"
+            ;
+
+        private static readonly Dictionary<string, string> normalizedOptions = new[]
+        {
+            AbortOnConnectFail,
+            AllowAdmin,
+            ClientName,
+            ConnectTimeout,
+            DefaultDatabase,
+            KeepAlive,
+            User,
+            Password,
+            Proxy,
+            ResponseTimeout,
+            Ssl,
+            Version,
+            SetClientLibrary,
+            Protocol,
+        }.ToDictionary(x => x, StringComparer.OrdinalIgnoreCase);
+
+        public static string TryNormalize(string value)
+        {
+            if (value != null && normalizedOptions.TryGetValue(value, out string? tmp))
+            {
+                return tmp ?? "";
+            }
+            return value ?? "";
+        }
+    }
+
+    private bool? ssl;
+
+    private Proxy? proxy;
+
+    private RetryStrategy? reconnectRetryPolicy;
+
+    private ReadFrom? readFrom;
+
+    /// <summary>
+    /// Gets or sets whether connect/configuration timeouts should be explicitly notified via a TimeoutException.
+    /// </summary>
+    public bool AbortOnConnectFail => true;
+
+    /// <summary>
+    /// Indicates whether admin operations should be allowed.
+    /// </summary>
+    public bool AllowAdmin => true;
+
+    /// <summary>
+    /// Specifies the time in milliseconds that the system should allow for asynchronous operations.
+    /// </summary>
+    /// <remarks>
+    /// Please use <see cref="ResponseTimeout" /> instead.
+    /// </remarks>
+    public int? AsyncTimeout => ResponseTimeout;
+
+    /// <summary>
+    /// Indicates whether the connection should be encrypted.
+    /// </summary>
+    [Obsolete("Please use .Ssl instead of .UseSsl."), Browsable(false), EditorBrowsable(EditorBrowsableState.Never)]
+    public bool UseSsl
+    {
+        get => Ssl;
+        set => Ssl = value;
+    }
+
+    /// <summary>
+    /// Gets whether the library should identify itself by library-name/version when possible.
+    /// </summary>
+    public bool SetClientLibrary => true;
+
+    /// <summary>
+    /// Gets the library name to use for CLIENT SETINFO lib-name calls to server during handshake.
+    /// </summary>
+    public string? LibraryName => "GlideC#";
```

Comment:
```suggestion
    public string? LibraryName => "GlideCSharp";
```
Based on list provided by Boaz in internal slack channel.